src/chigame/games/views.py

```diff
@@ -175,17 +190,34 @@ class TournamentUpdateView(UpdateView):
         "rules",
         "draw_rules",
         "num_winner",
-        "matches",
-        "players",
     ]
-    # Note: "winner" is not included in the fields because it is not
-    # supposed to be set by the user. It will be set automatically
-    # when the tournament is over.
-    # Note: we may remove the "matches" field later for the same reason,
-    # but we keep it for now because it is convenient for testing.
+
+    def dispatch(self, request, *args, **kwargs):
+        # Get the tournament object
+        tournament = self.get_object()
+
+        # Check if the current user is the creator of the tournament
+        if tournament.created_by != request.user and not request.user.is_staff:
+            raise PermissionDenied("You do not have permission to edit this tournament.")
+
+        # Continue with the normal flow
+        return super().dispatch(request, *args, **kwargs)
 
     def get_success_url(self):
-        return reverse_lazy("tournament-detail", kwargs={"pk": self.object.pk})
+        return reverse("tournament-detail", kwargs={"pk": self.object.pk})
+
+
+def distribute_tokens():
+    # Placeholder for future date check
+    # thirty_days_ago = datetime.now() - timedelta(days=30)
```

Comment:
Hmm we don't have to have a placeholder for future date check. We can just manually distribute tokens on a certain date every month to all users. Please see above link for a reference. Once this is addressed, we are good to go